modules/apps/transfer/types/msgs.go

```diff
@@ -106,11 +106,11 @@ func (msg MsgTransfer) ValidateBasic() error {
 		if err := msg.Forwarding.Validate(); err != nil {
 			return err
 		}
+	}
 
-		// We cannot have non-empty memo and non-empty forwarding hops at the same time.
-		if len(msg.Forwarding.Hops) > 0 && msg.Memo != "" {
-			return errorsmod.Wrapf(ErrInvalidMemo, "memo must be empty if forwarding path hops is not empty: %s, %s", msg.Memo, msg.Forwarding.Hops)
-		}
+	// We cannot have non-empty memo and non-empty forwarding hops at the same time.
+	if msg.ShouldBeForwarded() && msg.Memo != "" {
```

Comment:
It's a good question, actually. I suppose a transfer msg could be sent with an empty hop array (as the "starting" transfer) if someone created a new transfer message with "zero values" (i.e. not nil).

Reply:
I can add an item for discussion on monday and we can wait with this PR until then?

Reply:
It's probably better not to allow `forwarding != nil && len(forwarding.hops) == 0` because if the user has filled in a valid `forwarding.memo`, then the receiver chain will not execute any middleware triggered by the memo, because on `sendTransfer` we haven't coped the `forwarding.memo` to the regular `memo` field. Of course, we can handle that case in `sendTransfer` and make sure that if there are no hops we copy the memo, but maybe this sounds like fixing a user error that might or not be intended. Anyway, good to discuss it on Monday!

Reply:
Cleaned it up after your latest changes @DimitrisJim. Looks better now?

Reply:
Do you mean that `ShouldBeForwarded` only does a nil check and then we do the validation on the Forward itself? Yeah, that makes sense, actually! Good point 🙏

Reply:
Will clean this up once #6550 is done.

Reply:
Let me know if this looks good now @DimitrisJim